axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+The framework provides the `SpanAttributesProvider`, which can be registered to the `SpanFactory` either via its
+builder (if supported) or by calling the `SpanFactory.registerSpanAttributeProvider(provider)` method.
+
+The following `SpanAttributesProvider` implementations are included in Axon Framework:
+
+| Class                                       | label                       | description                                                                             |
+|:--------------------------------------------|-----------------------------|:----------------------------------------------------------------------------------------|
+| `AggregateIdentifierSpanAttributesProvider` | `axon_aggregate_identifier` | The aggregate identifier of the message, only present in case of a `DomainEventMessage` |
+| `MessageIdSpanAttributesProvider`           | `axon_message_id`           | The identifier of the message                                                           |
+| `MessageNameSpanAttributesProvider`         | `axon_message_name`         | The name of the message for Commands and Queries                                        |
+| `MessageTypeSpanAttributesProvider`         | `axon_message_type`         | The class of the message, such as `DomainEventMessage` or `GenericQueryMessage`         |
+| `PayloadTypeSpanAttributesProvider`         | `axon_payload_type`         | The class of the payload in the message                                                 |
+| `MetadataSpanAttributesProvider`            | `axon_metadata_{key}`       | All metadata of the message is also added to the span with its corresponding key        |
+
+In addition to the ones provided by the framework, you can also create a custom `SpanAttributesProvider`
+and add it to the `SpanFactory`, in case you want to add custom information you would like to include as a label.
+
+```java
+public class CustomSpanAttributesProvider implements SpanAttributesProvider {
+
+    @Nonnull
+    @Override
+    public Map<String, String> provideForMessage(@Nonnull Message<?> message) {
+        // Provide your labels based on the message here
+        return Collections.emptyMap();
+    }
+}
+```
+
+You can register this custom `SpanAttributesProvider` in one of the following ways.
+
+{% tabs %}
+{% tab title="Spring Boot Bean" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Bean
+    public SpanAttributesProvider customSpanAttributesProvider() {
+        // Autoconfiguration picks beans of type SpanAttributesProvider up automatically.
+        return new CustomSpanAttributesProvider();
+    }
+}
+```
+
+{% endtab %}
+{% tab title="Spring Boot injection" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Autowired
+    public void configureSpanFactory(SpanFactory spanFactory) {
+        spanFactory.registerSpanAttributeProvider(new CustomSpanAttributesProvider());
+    }
+}
+```
+
+{% endtab %}  
+{% tab title="Manual" %}
```

Comment:
Again I'd prefer the plain Axon configuration before any Axon Framework enhancements (like Spring).